File: cp_nim/src/bin/abc368-b-decrease-2-max-elements.nim
# Created by Ayush Biswas at 2026/07/19 10:35
# https://atcoder.jp/contests/adt_easy_20241001_2/tasks/abc368_b
import ../lib/cpio

# @code begin

import algorithm

var r = newReader()

var
  n = r[int]
  a = r[seq[int]].sorted(Descending)

var res: int
while a[1] > 0:
  a[0] -= 1
  a[1] -= 1
  inc res
  a.sort(Descending)

echo res
# @code end

File: cp_rust/src/bin/abc368-b-decrease-2-max-elements.rs
// Created by Ayush Biswas at 2026/07/10 22:17
// https://atcoder.jp/contests/adt_easy_20241001_2/tasks/abc368_b
use cp_lib::*;

// @code begin
use cpio::*;
use std::cmp::Reverse;

sol! {
    fn solution(
        _n: usize,
        (mut a): [usize]
    ) -> usize {
        let mut res = 0;
        loop {
            a.sort_by_key(|&x| Reverse(x));
            if a[1] == 0 {
                break;
            }
            a[0] -= 1;
            a[1] -= 1;
            res += 1;
        }
        res
    }
}
// @code end
